src/model_version_policy.hpp

```diff
@@ -95,6 +109,12 @@ class SpecificModelVersionPolicy : public ModelVersionPolicy {
      * @return Filtered version list
      */
     std::vector<model_version_t> filter(std::vector<model_version_t> versions) const override;
+
+    std::string conversion() const override {
+        std::stringstream versionStream;
+        std::copy(specificVersions.begin(), specificVersions.end(), std::ostream_iterator<model_version_t>(versionStream, " "));
+        return std::string("specific: ") + versionStream.str();
+    }
```

Comment:
Don't put implementation in header file if possible. It makes for longer compile times as each file including this header depends on dependencies from this implementation.

Reply:
What's more, instead of using `conversion()` function, you can implement cast operator:  
https://en.cppreference.com/w/cpp/language/cast_operator

```
operator std::string() const {
 // implementation

```
This way you can use it like this:
```
std::string converted = policy;
```